docker/docker.go

```diff
@@ -0,0 +1,131 @@
+package docker
+
+// this package recieves the Docker Hub Automated Build webhook
+// https://docs.docker.com/docker-hub/webhooks/
+// NOT the Docker Trusted Registry webhook
+// https://docs.docker.com/ee/dtr/user/create-and-manage-webhooks/
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// parse errors
+var (
+	ErrInvalidHTTPMethod = errors.New("invalid HTTP Method")
+	ErrParsingPayload    = errors.New("error parsing payload")
+)
+
+// Event defines a GitHub hook event type
+type Event string
+
+// GitHub hook types
+const (
+	BuildEvent Event = "build"
+)
+
+// BuildPayload a docker hub build notice
+// https://docs.docker.com/docker-hub/webhooks/
+type BuildPayload struct {
+	CallbackURL string `json:"callback_url"`
+	PushData    struct {
+		Images   []string `json:"images"`
+		PushedAt float32  `json:"pushed_at"`
+		Pusher   string   `json:"pusher"`
+		Tag      string   `json:"tag"`
+	} `json:"push_data"`
+	Repository struct {
+		CommentCount    int     `json:"comment_count"`
+		DateCreated     float32 `json:"date_created"`
+		Description     string  `json:"description"`
+		Dockerfile      string  `json:"dockerfile"`
+		FullDescription string  `json:"full_description"`
+		IsOfficial      bool    `json:"is_official"`
+		IsPrivate       bool    `json:"is_private"`
+		IsTrusted       bool    `json:"is_trusted"`
+		Name            string  `json:"name"`
+		Namespace       string  `json:"namespace"`
+		Owner           string  `json:"owner"`
+		RepoName        string  `json:"repo_name"`
+		RepoURL         string  `json:"repo_url"`
+		StarCount       int     `json:"star_count"`
+		Status          string  `json:"status"`
+	} `json:"repository"`
+}
+
+// there are no options for docker webhooks
+// however I'm leaving this here for now in anticipation of future support for Docker Trusted Registry
```

Comment:
Can we remove this and all the other commented out code below :)

I appreciate you looking ahead, but we can easily copy this back from any of the other webhook in this package.